runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientFactory.java

```diff
@@ -62,12 +63,20 @@ public KafkaClientFactory(
         final BindingHandler streamFactory = config.clientConnectionPool() ? connectionPool.streamFactory() :
                 context.streamFactory();
 
+        final Function<Integer, BindingHandler> streamFactorySupplier = d ->
```

Comment:
Avoid auto-boxing, use `IntFunction<BindingHandler>` instead, for all references too.